src/globus_sdk/scopes/_parser.py

```diff
@@ -233,6 +239,24 @@ def _convert_trees(trees: list[ScopeTreeNode]) -> ScopeGraph:
     return graph
 
 
+def _peek_enumerate(data: str | list[str]) -> t.Iterator[tuple[int, str, str | None]]:
+    """
+    An iterator producing (index, character, next_char)
+    or else producing (index, str, next_str)
+
+    (Depending on whether or not the input is a string or list of strings)
+    """
+    if not data:
+        return
+
+    prev: str = data[0]
+    for idx, c in enumerate(data[1:]):
+        yield (idx, prev, c)
+        prev = c
+
+    yield (len(data) - 1, prev, None)
```

Comment:
It's gonna be so nice when Python 3.10 is the lowest supported version and we can use `itertools.pairwise()`!